File: _practice/539-Minimum-Time-Difference/neet.js
// total sam

var findMinDifference = function (timePoints) {
  const toMinutes = (time) => {
    const [hours, minutes] = time.split(":").map(Number);
    return hours * 60 + minutes;
  };

  timePoints.sort((a, b) => toMinutes(a) - toMinutes(b));

  // initialize result with time difference of first and last points if we go counter clockwise
  // this will be the only case we need to compute counter clockwise difference
  let res =
    24 * 60 -
    toMinutes(timePoints[timePoints.length - 1]) +
    toMinutes(timePoints[0]);

  for (let i = 0; i < timePoints.length - 1; i++) {
    const cur = toMinutes(timePoints[i + 1]);
    const prev = toMinutes(timePoints[i]);
    res = Math.min(res, cur - prev);
  }

  return res;
};

File: _practice/539-Minimum-Time-Difference/neet.py
# total sam

class Solution:
    def findMinDifference(self, timePoints) -> int:
        timePoints.sort()

        def time_to_minutes(t):
            h, m = map(int, t.split(":"))
            return 60 * h + m
        
        res = 24 * 60 - time_to_minutes(timePoints[-1]) + time_to_minutes(timePoints[0]) # compute circular distance between first and last points
        for i in range(len(timePoints) - 1):
            cur = time_to_minutes(timePoints[i + 1])            
            prev = time_to_minutes(timePoints[i])
            diff = cur - prev
            res = min(res, diff)

        return res    